adapters/handlers/rest/authz/validation.go

```diff
@@ -0,0 +1,86 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package authz
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/schema"
+)
+
+var errCollectionRequired = fmt.Errorf("collection is required")
+
+func validatePermissions(permissions ...*models.Permission) error {
+	if len(permissions) == 0 {
+		return fmt.Errorf("role has to have at least 1 permission")
+	}
+
+	for _, perm := range permissions {
+		var multiErr error
+		if perm.Collections != nil {
+			if perm.Collections.Collection == nil {
+				return errCollectionRequired
+			}
+			_, err := schema.ValidateClassNameIncludesRegex(*perm.Collections.Collection)
+			multiErr = errors.Join(err)
+		}
+
+		if perm.Tenants != nil {
+			if perm.Tenants.Collection == nil {
+				return errCollectionRequired
+			}
+
+			_, classErr := schema.ValidateClassNameIncludesRegex(*perm.Tenants.Collection)
+			multiErr = errors.Join(classErr)
+
+			if perm.Tenants.Tenant != nil {
+				multiErr = errors.Join(schema.ValidateTenantNameIncludesRegex(*perm.Tenants.Tenant))
+			}
```

Comment:
does this mean that we require a collection to be send, but not a tenant? Why would we handle that different?

Reply:
i have pushed a change